### ISSUE TITLE
[idea] Use second factor as challenge

### ISSUE THREAD

**Issue by Vincent43** [eworm-de/mkinitcpio-ykfde#20]
Using second factor (user password) as challenge will allow for not storing challenge on disk and keep it secret. It would need asking for second factor before  challenge-responese action.

**Comment by eworm-de** [comment]
This already supports second factor on top of the stored challenge. What's wrong with that?

**Comment by Vincent43** [comment]
Challenge is stored in clear available to anyone. If someone has access to yubikey, the response and at least half of luks passphrase is avalaible . That means it works same as static password stored in yubikey. If second factor is used as challenge access to yubikey alone gives nothing. Also stored challenge doesn't have any useful purpose if second factor is needed anyway and can be used as challenge instead.

So I propose that stored challenge will be used only in one factor case.

For two factor user password will be used as both challenge and second half of luks passphrase.

**Comment by eworm-de** [comment]
> Challenge is stored in clear available to anyone. If someone has access to yubikey, the response and at least half of luks passphrase is avalaible . That means it works same as static password stored in yubikey.

No. Because you need Yubikey and challenge at the same time and place.

> If second factor is used as challenge, access to yubikey alone gives nothing. Also stored challenge doesn't have any useful purpose if second factor is needed anyway and can be used as challenge instead.

Wrong. The challenge (and LUKS passphrase) is changed on every boot. So consider the stored challenge is a dynamic salt: If anybody manages to calculate the response (aka LUKS passphrase) it is invalid on next boot.

> So I propose that stored challenge will be used only in one factor case.

No. See above.

> For two factor user password will be used as both challenge and second half of luks passphrase.

That does not bring any benefit but extra code complexity.

**Comment by PeterFeicht** [comment]
I know this is an old issue, but from what I can see this is implemented already. The static challenge and second factor (e.g. from user input) are combined to form the challenge to the YubiKey.
So I think this can be closed.
